media/server/main/source/MediaKeysServerInternal.cpp

```diff
@@ -691,4 +691,49 @@ void MediaKeysServerInternal::ping(std::unique_ptr<IHeartbeatHandler> &&heartbea
 
     m_mainThread->enqueueTaskAndWait(m_mainThreadClientId, task);
 }
+
+MediaKeyErrorStatus MediaKeysServerInternal::getMetricSystemData(std::vector<uint8_t> &buffer)
+{
+    RIALTO_SERVER_LOG_DEBUG("entry:");
+
+    uint32_t bufferLength{1024};
+    const uint32_t kMaxBufferLength{65536};
+    MediaKeyErrorStatus status;
+    buffer.resize(bufferLength);
+
+    for (int attempts = 0; bufferLength <= kMaxBufferLength; ++attempts)
+    {
+        auto task = [&]() { status = m_ocdmSystem->getMetricSystemData(bufferLength, buffer); };
+        m_mainThread->enqueueTaskAndWait(m_mainThreadClientId, task);
+
+        if (status != MediaKeyErrorStatus::BUFFER_TOO_SMALL)
+        {
+            RIALTO_SERVER_LOG_ERROR("Buffer too small with a size of %u", bufferLength);
+            break;
+        }
+
+        if (bufferLength >= kMaxBufferLength)
+        {
+            RIALTO_SERVER_LOG_ERROR("Buffer size %u exceeds the maximum allowed size %u", bufferLength, kMaxBufferLength);
+            return MediaKeyErrorStatus::BUFFER_TOO_SMALL;
+        }
+
+        RIALTO_SERVER_LOG_WARN("Buffer is too small, resizing from %u to %u", bufferLength, bufferLength * 2);
+        bufferLength *= 2;
+        buffer.resize(bufferLength);
+    }
+
+    if (status == MediaKeyErrorStatus::OK)
+    {
+        buffer.resize(bufferLength);
```

Comment:
I would add a comment why this resize is needed, because it might not be obvious